cdc/puller/ddl_puller.go

```diff
@@ -561,6 +582,28 @@ func NewDDLJobPuller(
 		outputCh:      make(chan *model.DDLJobEntry, defaultPullerOutputChanSize),
 	}
 	if jobPuller.multiplexing {
+		mp := &jobPuller.multiplexingPuller
+
+		rawDDLCh := make(chan *model.RawKVEntry, defaultPullerOutputChanSize)
+		mp.sortedDDLCh = memorysorter.SortOutput(ctx, changefeed, rawDDLCh)
+
+		client := kv.NewSharedClient(
+			changefeed, cfg, ddlPullerFilterLoop,
+			pdCli, grpcPool, regionCache, pdClock,
+			txnutil.NewLockerResolver(kvStorage.(tikv.Storage), changefeed),
+		)
+		consume := func(ctx context.Context, raw *model.RawKVEntry, _ []tablepb.Span) error {
```

Comment:
It is recommended to retain the previous output channel, which can reduce the coupling degree of different modules.

Reply:
It's possible that the output is not a channel. In SourceManager, `consume` will write events into `SortEngine` directly.